File: Backend/INMS.Domain/Entities/Alarm.cs
namespace INMS.Domain.Entities;

public class Alarm
{
    public int AlarmId { get; set; }
    public int DeviceId { get; set; }
    public string AlarmType { get; set; }
    public DateTime RaisedTime { get; set; }
    public DateTime? ClearedTime { get; set; }
    public bool IsActive { get; set; }
}

File: Frontend-INMS/src/models/Alarm.js
/**
 * @typedef {Object} Alarm
 * @property {number} alarmId
 * @property {number} deviceId
 * @property {string} alarmType
 * @property {string} raisedTime
 * @property {string} [clearedTime]
 * @property {boolean} isActive
 */

// This file serves as documentation for the Alarm model
export const alarmModelSchema = {
  alarmId: 'number',
  deviceId: 'number',
  alarmType: 'string',
  raisedTime: 'string (ISO datetime)',
  clearedTime: 'string (ISO datetime, optional)',
  isActive: 'boolean'
};
